docker-compose.yml

```diff
@@ -11,16 +11,16 @@ services:
     tmpfs: ['/tmp']
     depends_on: 
       - db
+    env_file:
+      - .env
     environment:
       GIN_MODE: release
-      ASG_ID: ""
-      ASG_SECRET: ""
-      VAULT_ADDR: ""
-      VAULT_TOKEN: ""
+      VAULT_ADDR: ${VAULT_ADDR}
+      VAULT_TOKEN: ${VAULT_TOKEN}
```

Comment:
Is the `.env` supposed to contain `VAULT_ADDR` and `VAULT_TOKEN`, since you did mention:
"`VAULT_ADDR` and `VAULT_TOKEN` in an `.env` file."

If so, these two should not be set by host env var.

Reply:
Yes, the user can add it in the `.env` file if so inclined or just use the env var in the shell - which would be common for `VAULT_ADDR` and `VAULT_TOKEN`, so i left these two in (whereas `ASG_ID` and `ASG_SECRET` is less common)

DO you think it's better that i edit the README then?

Reply:
I think there will be an issue if you assign `environment` (instead of only from `env_file`) here. 

The environment always takes precedence, rendering the env file useless; so the effective values are always from the host, i.e. the host must always set `VAULT_ADDR` and `VAULT_TOKEN` in their local set-up.

You can try it, `unset` the two above env vars from your system and set only in `.env`, you should see errors.

If the above is correct, my suggestion is to drop the `environment`, and just rely on `.env` (please add to `.gitignore` this env path after that).

Reply:
You are right that `environment` always take precedence [0], but i wasn't sure if `environment` would fall back to the default in `env_file` if the env var is missing in the shell. So i tried out your scenario and it does fall back to `env_file` if the env var is missing, so the setup works. Thanks for bringing this up.

[0] https://docs.docker.com/compose/environment-variables/#the-env_file-configuration-option
When you run the container, the environment variable defined in the Compose file takes precedence.

Reply:
Interesting, didn't know if you don't define the env vars in the host, it actually falls back to something else (here it is the `env_file`) that has the env vars defined.

Then in that case it is fine, thanks for trying it out!